apps/hash-frontend/src/shared/account-pages-variables.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  currentTimeInstantTemporalAxes,
+  generateVersionedUrlMatchingFilter,
+  zeroedGraphResolveDepths,
+} from "@local/hash-isomorphic-utils/graph-queries";
+import { systemTypes } from "@local/hash-isomorphic-utils/ontology-types";
+import { OwnedById } from "@local/hash-subgraph";
+
+import { notArchivedFilter } from "../pages/shared/not-archived-filter";
+
+export const getAccountPagesVariables = ({
+  ownedById,
+  includeArchived = false,
+}: {
+  ownedById?: OwnedById;
+  includeArchived?: boolean;
+}) => ({
+  query: {
+    filter: {
+      all: [
+        generateVersionedUrlMatchingFilter(
+          systemTypes.entityType.page.entityTypeId,
+          // ignoreParents assumes we don't have types which are children of Page which should be returned here
+          { ignoreParents: true },
+        ),
+        {
+          equal: [{ path: ["ownedById"] }, { parameter: ownedById }],
+        },
+        ...(includeArchived ? [] : [notArchivedFilter]),
+      ],
+    },
+    graphResolveDepths: {
+      ...zeroedGraphResolveDepths,
+      isOfType: { outgoing: 1 },
+      hasLeftEntity: { incoming: 1, outgoing: 0 },
+      hasRightEntity: { incoming: 0, outgoing: 1 },
```

Comment:
Maybe we can add a comment for what we're fetching here, I assume it's the parents of the page:
```suggestion
      // These depths are chosen to cover the following:
      //  - the page's parent page (page -> [hasLeftEntity incoming 1] parent [hasRightEntity outgoing 1] -> page)
      hasLeftEntity: { incoming: 1, outgoing: 0 },
      hasRightEntity: { incoming: 0, outgoing: 1 },
```